### PR TITLE
Add a scale selector example

### PR DESCRIPTION
**Pull request by elemoine**
This PR adds a scale selector example. This example creates a "scale selector" directive based on Bootstrap's `dropdown` jQuery plugin. The view resolution changes when a scale is selected in the scale selector. And the selected scale changes in the scale selector when the view resolution changes.

See http://erilem.net/ngeo/scaleselector/scaledropdown.html.

### CODE REVIEW

**Comment by asaunier** [comment]
Thanks, it looks good.

What about adding a scaleselector directive in ngeo, not only an example? This would save people from writing their own directive, even if an example is provided.

Also I have found the scaleselector controller code difficult to read because it uses many functions that call each other along with event handlers. Isn't it possible to simplify it a bit? :P

File: examples/scaledropdown.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    '2': $sce.trustAsHtml('1&nbsp;:&nbsp;50\'000\'000'),
+    '3': $sce.trustAsHtml('1&nbsp;:&nbsp;25\'000\'000'),
+    '4': $sce.trustAsHtml('1&nbsp;:&nbsp;12\'000\'000')
+  };
+
+  /**
+   * @type {ol.Map}
+   */
+  var map = this['map'];
+  window.map = map;
```
Comment (elemoine) [inline examples/scaledropdown.js]: Remove this.

File: examples/scaledropdown.html [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          content="initial-scale=1.0, user-scalable=no, width=device-width">
+    <meta name="apple-mobile-web-app-capable" content="yes">
+    <link rel="stylesheet" href="../node_modules/openlayers/css/ol.css" type="text/css">
+    <link rel="stylesheet" href="../node_modules/bootstrap/dist/css/bootstrap.css" type="text/css">
+    <style>
+      #map {
+        height: 400px;
+      }
+      .dropdown button {
+        min-width: 100px;
```
Comment (elemoine) [inline examples/scaledropdown.html]: Make it bigger.

**Comment by elemoine** [comment]
> What about adding a scaleselector directive in ngeo, not only an example? This would save people from writing their own directive, even if an example is provided.

It's a bit too specific to my taste at this point, because it relies on Bootstrap's dropdown plugin. See the background layer directive for examples. There are two examples, one based on the native `select` element, and one based on Bootstrap's `dropdown` plugin. Which one would you add to ngeo?

**Comment by elemoine** [comment]
> Also I have found the scaleselector controller code difficult to read because it uses many functions that call each other along with event handlers. Isn't it possible to simplify it a bit? :P

I declare today that anyone requesting more simplicity should go ahead and try to implement the thing by himself or herself :) More seriously, I did it this way to please the compiler, and I don't find it too complex personally. Do you have concrete suggestions?

**Comment by asaunier** [comment]
> Do you have concrete suggestions?

Of course not! :)

**Comment by elemoine** [comment]
> It's a bit too specific to my taste at this point, because it relies on Bootstrap's dropdown plugin. See the background layer directive for examples. There are two examples, one based on the native select element, and one based on Bootstrap's dropdown plugin. Which one would you add to ngeo?

That being said I agree with you that having something in the library would be better. I'll think more about it.

**Comment by elemoine** [comment]
PR updated, I created an `ngeoScaleselector` directive, as suggested by @asaunier.

**Comment by asaunier** [comment]
> I created an ngeoScaleselector directive

Great, thanks!

**Comment by elemoine** [comment]
Thanks @asaunier.

I've just modified the directive to make it possible to use the dropup variation of the Bootstrap dropdown plugin.

CC @pgiraud.

**Comment by elemoine** [comment]
> I've just modified the directive to make it possible to use the dropup variation of the Bootstrap dropdown plugin.

Also demonstrate how we can pass pass options objects to directives. We may want to apply that principle to other directives in the future (for optional options).

**Comment by asaunier** [comment]
> Also demonstrate how we can pass pass options objects to directives.

Yes, that's a good example!

**Comment by elemoine** [comment]
I've made another change: the options object, instead of just the `dropup` option value, is now made available to the partial. In this way, user-specific options can be passed and used in a user-specific partial.

Thanks for the review. I'll merge this when the Travis build passes.

**Pull request merged**